src/components/UserContributionForm.vue

```diff
@@ -0,0 +1,130 @@
+<script setup>
+import { ref, computed } from "vue";
+import { defineProps } from "vue";
+import { recordUserContribution } from "../stores/api.js";
+
+const data = defineProps({
+  description: String,
+  inputOptions: Object,
+  missingFieldName: String,
+  toolName: String,
+  taskId: Number,
+  isError: Boolean,
+  getNextTask: Function,
+});
+
+const inputOptionsArray = computed(() => {
+  return data?.inputOptions
+    ? Object.entries(data.inputOptions).map(([key, value]) => ({ key, value }))
+    : [];
+});
+
+const missingFieldValue = ref(null);
+const errorMessage = ref("");
+
+const submit = async () => {
+  const fieldValue = missingFieldValue.value;
+  const contributionRecord = {
+    value: Array.isArray(fieldValue) ? fieldValue.join() : fieldValue,
+    field: data.missingFieldName,
+    tool: data.toolName,
+  };
+
+  try {
+    const res = await recordUserContribution(data.taskId, contributionRecord);
+    alert(res)
+  } catch (error) {
+    console.log(error);
```

Comment:
Can we also have a "Failed" message around the input field if any errors. Also if the error is due to the user not being logged in, we should be able to have a message to notify them to do so.